:100644 100644 cb1f7b7 0000000 M	libavcodec/dca_core.c

diff --git a/libavcodec/dca_core.c b/libavcodec/dca_core.c
index cb1f7b7..7840d58 100644
--- a/libavcodec/dca_core.c
+++ b/libavcodec/dca_core.c
@@ -2372,7 +2372,7 @@ int ff_dca_core_filter_frame(DCACoreDecoder *s, AVFrame *frame)
     // Set profile, bit rate, etc
     if (s->ext_audio_mask & DCA_EXSS_MASK)
         avctx->profile = AV_PROFILE_DTS_HD_HRA;
-    else if (s->ext_audio_mask & (DCA_CSS_XXCH | DCA_CSS_XCH))
+    else if (s->es_format || s->ext_audio_mask & (DCA_CSS_XXCH | DCA_CSS_XCH))
         avctx->profile = AV_PROFILE_DTS_ES;
     else if (s->ext_audio_mask & DCA_CSS_X96)
         avctx->profile = AV_PROFILE_DTS_96_24;
